### PR TITLE
Questionnaires in activities and tasks

### LINKED ISSUE #28: Questionnaires fixes & improvements
Related to PR #26 

Admin:

- [x] Add migrations to add questionnaire to existing Activities, Tasks
- [x] Change admin icon for questionnaire
- [x] Remove custom questionnaires from activities, lets focus on Task questionnaires only. This will facilitate the creation of graphs where to compare different activities
- [ ] Give the admin a way to go back when editing questionnaire
- [x] Allow users to answer the same questionnaire if activities are different
- [x] Return to task list after saving questionnaire
- [ ] Pre-populate a template with a simple questionnaire - superseed by #38
    - 2 questions: what reputation has this task (from 1 to 10)? what is your perception about this task, who does this task usually: women, man, equal, N/A
- [ ] Pre-populate terms and services? - superseed by #38
- [ ] Add questionnaire seeds - superseed by #38

Front:

- [x] When a task has a questionnaire, it should appear a button to answer it (see doubts)
- [x] Use another endpoint than `join` to answer the questionnaire
- [x] Check the behavior (and fix) when user is answering the questionnaire

Doubts:

- Should we make mandatory to answer a questionnaire when joining an activity?

### PR DESCRIPTION
**Pull request by davidbeig**
This PR will add questionnaires to the models of activity and tasks. This way the administrator will be able to create questionnaires so that the user answers them before joining a new activity/task.
Fixes #28

### CODE REVIEW

File: app/controllers/decidim/time_tracker/activities_questionnaire_controller.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module Decidim
+  module TimeTracker
+    class ActivitiesQuestionnaireController < ApplicationController
+      include Decidim::Forms::Concerns::HasQuestionnaire
+
+      def answer; end
```
Comment (davidbeig) [inline app/controllers/decidim/time_tracker/activities_questionnaire_controller.rb:8]: Here I'm missing how to handle the answer of the questionnaire. I guess that after saving the answer model, the user has to be redirected to requests_path(activity) (as defined in the time_tracker_controller). I still don't see the correct flow that the has_questionnaire concert follows when answering a questionnaire.

File: app/views/decidim/time_tracker/admin/activities/index.html.erb [hunk 2]
```diff
@@ -48,6 +48,10 @@
                   <%= icon_link_to "pencil", edit_task_activity_path(@current_task, activity), t("actions.edit", scope: "decidim.time_tracker"), class: "action-icon--edit" %>
                 <% end %>
 
+                <% if allowed_to? :update, :activity, activity: activity %>
+                  <%= icon_link_to "people", edit_task_activity_form_path(@current_task, activity), t("actions.questionnaire", scope: "decidim.time_tracker"), class: "action-icon--edit" %>
```
Comment (davidbeig) [inline app/views/decidim/time_tracker/admin/activities/index.html.erb:52]: We need a better icon than "people"

File: lib/decidim/time_tracker/engine.rb [hunk 3]
```diff
@@ -16,6 +16,11 @@ class Engine < ::Rails::Engine
           resources :activities do
             post :start, controller: "time_events"
             post :stop, controller: "time_events"
+
+            resource :join, controller: "activities_questionnaire"
```
Comment (davidbeig) [inline lib/decidim/time_tracker/engine.rb:20]: I have named "join" the resource used to answer the questionnaire, but I still don't manage to know if I should limit the endpoints using "only" as well as how to define the correct endpoints used to answer the questionnaires

File: app/views/decidim/time_tracker/admin/activities/index.html.erb (hunk 2 above)
Comment (verarojman) [reply app/views/decidim/time_tracker/admin/activities/index.html.erb:52]: What about `document` ?
image

Comment (microstudi) [reply app/views/decidim/time_tracker/admin/activities/index.html.erb:52]: there aso 
image

File: config/locales/en.yml [hunk 4]
```diff
@@ -221,6 +225,9 @@ en:
           inactive: Inactive
           not_started: Will start on %{start_date}
           open: Open
+          questionnaire:
+            answered: Thanks for anwering the questionnaire!
```
Comment (verarojman) [inline config/locales/en.yml:229]:
```suggestion
            answered: Thanks for answering the questionnaire!
```

File: config/locales/en.yml [hunk 5]
```diff
@@ -50,19 +50,15 @@ en:
...
-        edit: Edit
-        new: New
-        reject: Reject
-        title: Actions
       admin:
+        actions:
+          title: Actions
         activities:
           actions:
+            assignees: Edit assignees
```
Comment (verarojman) [inline config/locales/en.yml:58]:
```suggestion
            assignees: Manage assignees
```

**Review by verarojman — CHANGES_REQUESTED** [review]
Awesome! Just a typo and a suggestion in the locales.

**Review by verarojman — APPROVED** [review]

**Pull request merged**